pkg/parse/parse.go

```diff
@@ -414,6 +428,130 @@ func GetBindMount(args []string, contextDir string) (specs.Mount, error) {
+	var err error
+	newMount := specs.Mount{
+		Type: TypeBind,
+	}
+
+	setDest := false
+	bindNonRecursive := false
+	setShared := false
+	setReadOnly := false
+	// if id is set a new subdirectory with `id` will be created under /host-temp/buildah-build-cache/id
+	id := ""
+	//buidkit parity: cache directory defaults to 755
+	mode := 0755
+	//buidkit parity: cache directory defaults to uid 0 if not specified
+	uid := 0
+	//buidkit parity: cache directory defaults to gid 0 if not specified
+	gid := 0
+
+	for _, val := range args {
+		kv := strings.SplitN(val, "=", 2)
+		switch kv[0] {
+		case "bind-nonrecursive":
+			newMount.Options = append(newMount.Options, "bind")
+			bindNonRecursive = true
+		case "nosuid", "nodev", "noexec":
+			// TODO: detect duplication of these options.
+			// (Is this necessary?)
+			newMount.Options = append(newMount.Options, kv[0])
+		case "rw", "readwrite":
+			newMount.Options = append(newMount.Options, "rw")
+		case "readonly", "ro":
+			// Alias for "ro"
+			newMount.Options = append(newMount.Options, "ro")
+			setReadOnly = true
+		case "shared", "rshared", "private", "rprivate", "slave", "rslave", "Z", "z", "U":
+			newMount.Options = append(newMount.Options, kv[0])
+			setShared = true
+		case "bind-propagation":
+			if len(kv) == 1 {
+				return newMount, errors.Wrapf(optionArgError, kv[0])
+			}
+			newMount.Options = append(newMount.Options, kv[1])
+		case "id":
+			id = kv[1]
+		case "target", "dst", "destination":
+			if len(kv) == 1 {
+				return newMount, errors.Wrapf(optionArgError, kv[0])
+			}
+			if err := parse.ValidateVolumeCtrDir(kv[1]); err != nil {
+				return newMount, err
+			}
+			newMount.Destination = kv[1]
+			setDest = true
+		case "mode":
+			mode, err = strconv.Atoi(kv[1])
+			if err != nil {
+				return newMount, errors.Errorf("Unable to parse cache mode: %v", err)
+			}
+		case "uid":
+			uid, err = strconv.Atoi(kv[1])
+			if err != nil {
+				return newMount, errors.Errorf("Unable to parse cache uid: %v", err)
+			}
+		case "gid":
+			gid, err = strconv.Atoi(kv[1])
+			if err != nil {
+				return newMount, errors.Errorf("Unable to parse cache gid: %v", err)
+			}
+		default:
+			return newMount, errors.Wrapf(errBadMntOption, kv[0])
+		}
+	}
+
+	if !setDest {
+		return newMount, noDestError
+	}
+
+	// since type is cache and cache can be resused by consecutive builds
+	// create a common cache directory, which should persists on hosts within temp lifecycle
+	// add subdirectory if specified
+	if id != "" {
+		newMount.Source = filepath.Join(os.TempDir(), BuildahCacheDir, id)
+	} else {
+		newMount.Source = filepath.Join(os.TempDir(), BuildahCacheDir)
+	}
+	// create cache on host if not present
+	err = os.MkdirAll(newMount.Source, os.FileMode(mode))
+	if err != nil {
+		return newMount, errors.Errorf("Unable to create build cache directory: %v", err)
+	}
+	//buidkit parity: change uid and gid if specificed otheriwise keep `0`
+	err = os.Chown(newMount.Source, uid, gid)
```

Comment:
Do you need to do anything about SELinux handling here?

Reply:
`Z` is already supported. So users are free to use `z` if they need selinux label but its not added by default.